src/libs/actions/App.js

```diff
@@ -45,29 +45,29 @@ Onyx.connect({
     },
 });
 
-let policyIDList = [];
+let allPolicies = [];
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.POLICY,
     waitForCollectionCallback: true,
     callback: (policies) => {
-        policyIDList = _.compact(_.pluck(policies, 'id'));
+        allPolicies = policies;
     },
 });
 
-// When we reconnect the app, we don't want to fetch workspaces created optimistically while offline since they don't exist yet in the back-end.
-// If we fetched them then they'd return as empty objects which would clear out the optimistic policy values initially created locally.
-// Once the re-queued call to CreateWorkspace returns, the full contents of the workspace excluded here should be correctly saved into Onyx.
-let policyIDListExcludingWorkspacesCreatedOffline = [];
-Onyx.connect({
-    key: ONYXKEYS.COLLECTION.POLICY,
-    waitForCollectionCallback: true,
-    callback: (policies) => {
-        const policiesExcludingWorkspacesCreatedOffline = _.reject(policies,
-            policy => lodashGet(policy, 'pendingAction', null) === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD
-            && lodashGet(policy, 'type', null) === CONST.POLICY.TYPE.FREE);
-        policyIDListExcludingWorkspacesCreatedOffline = _.compact(_.pluck(policiesExcludingWorkspacesCreatedOffline, 'id'));
-    },
-});
+/**
+ * When we reconnect the app, we don't want to fetch workspaces created optimistically while offline since they don't exist yet in the back-end.
+ * If we fetched them then they'd return as empty objects which would clear out the optimistic policy values initially created locally.
+ * Once the re-queued call to CreateWorkspace returns, the full contents of the workspace excluded here should be correctly saved into Onyx.
```

Comment:
I'm not sure that this description makes sense to me. When I read this:

> If we fetched them then they'd return as empty objects

The first thing I think is... why don't we just update the API to not return them as empty objects?

> Once the re-queued call to CreateWorkspace returns

I'm not really sure what this means. What does "re-queue" mean in this context and why does it happen for CreateWorkspace? What does "returns" mean (return is a statement at the end of a method)?

A lot of this comment is assuming a lot about network requests, and other things happening in Onyx, yet the code inside the method has nothing to do with any of that. The code inside the method is just filtering policies. I would suggest simply renaming this method to `getNonOptimisticPolicyIDs` and drop the method description entirely. With a name like that, it's clear what the method does.

Reply:
> I'm not sure that this description makes sense to me. When I read this:
> 
> > If we fetched them then they'd return as empty objects
> 
> The first thing I think is... why don't we just update the API to not return them as empty objects?

Yeah, the comment seems to be wrong about this, the API doesn't return them as "empty object" it creates Onyx updates setting them to `null`.  This code: https://github.com/Expensify/Web-Expensify/blob/383a1bff0220bf2b795715f371ad443dc634283c/lib/AppInit.php#L87-L103

> 
> > Once the re-queued call to CreateWorkspace returns
> 
> I'm not really sure what this means. What does "re-queue" mean in this context and why does it happen for CreateWorkspace? What does "returns" mean (return is a statement at the end of a method)?
> 
> A lot of this comment is assuming a lot about network requests, and other things happening in Onyx, yet the code inside the method has nothing to do with any of that. The code inside the method is just filtering policies. I would suggest simply renaming this method to `getNonOptimisticPolicyIDs` and drop the method description entirely. With a name like that, it's clear what the method does.

Agreed!

Reply:
Dropped the long description